text/204-golang-bindings.md

```diff
@@ -0,0 +1,669 @@
+}
+
+// Generated interface for Cluster class
+type ClusterIface interface {
+    ICluster
+}
+
+// Generated struct for Cluster class
+type Cluster struct {
+    Resource
+    ICluster
+
+    connections ec2.Connections
+    vpc         ec2.IVpc
+    clusterArn  string
+    clusterName string
+
+    defaultCloudMapNamespace cloudmap.INamespace
+    hasEc2Capacity           bool
+    autoscalingGroup         autoscaling.IAutoScalingGroup
+}
+
+func (c *Cluster) ClusterName() string {
+    return c.clusterName
+}
+
+// ...etc
+```
+
+### Case 4: Abstract Classes
+
+These should be able to be handled much in the same way as regular classes, by generating an interface and a struct. The struct generation is still
+necessary in this case because method implementations can actually be defined on abstract classes in Typescript.
+
+## Other considerations
+
+### Primitive types
+
+#### Date
+
+JSII Date would be satisfied by Go's `time.Time` type.
+
+#### String
+
+JSII String would be satisfied by Go's string type.
+
+#### Number
+
+JSII number does not distinguish the difference between integer and float. This either requires JSII for Go to treat all numbers as float64, or define
+a new jsii.Number type that "wraps" Go number types into the singular JSII Number type.
+
+Typescript numbers are float64 with a max integer size of 2^53 -1. This means that Customers could want to provide a int64 value that would lose
+precision.
+
+#### Any
+
+JSII any maps to Go's empty interface type. JSII for Go should replace all JSII any types with interface{}.
+
+### Promise
+
+Promises do not exist natively in Go. JSII functions would be wrapped within a synchronous Go function that wait for the async JSII function to be
+complete, and return either the result or an error. This is the same pattern used for Java and .Net JSII bindings.
+
+```go
+func JSIIPromise() (resp, error) { /* ... */ }
+```
+
+### Packages and submodules
+
+Go package names correspond to a single directory, since their full path is used when imported from other packages. Packages cannot be split across
+multiple directories, nor can more than one package be specified in the same directory. NOTE: This does not apply subdirectories.
+
+By convention, package names in go are lower case without underscores or mixed caps. However, many of the module names in the CDK are quite
+long/verbose, resulting in names that don’t necessarily fit the succinct style of Go. Currently, we are staying with convention of having a single,
+lower-cased package name. However, we may want to consider alternatives such as adding underscores or mixed caps for readability.
+
+### Packaging
+
+Unlike Node, which has NPM, or other languages, there is no central package repository for Go; packages/dependencies are fetched from their source
+location, usually on GitHub.
+
+Packages are imported by name (for stdlib packages) or URL (for external packages) using the import keyword, and should be grouped alphabetically
```

Comment:
I believe from seeing headlines pass by that there is some bleeding-edge work going on here at the moment, no? Could we benefit from subscribing to some new developments in this area?

Reply:
Golang doesn't have a central package repository.

They recently'ish introduced go modules which allow you to manage the versions of your packages and they introduced a proxy to improve speed and security. you can find more information here https://proxy.golang.org/